diffusion/datasets/coco/coco_captions.py

```diff
@@ -22,23 +22,21 @@ class StreamingCOCOCaption(StreamingDataset):
             Default: ``None``.
         local (str, optional): Local filesystem directory where dataset is cached during operation.
             Default: ``None``.
-        split (str, optional): The dataset split to use. Currently, only ``None`` is supported.
-            Default: ``None``.
         shuffle (bool): Whether to shuffle the samples in this dataset.
             Default: ``False``.
+        shuffle_algo (str): What shuffle algorithm to use.
+            Default: ``'py1s'``.
```

Comment:
Out of curiosity, why we do want to use `py1s` is there any reason to?

Reply:
We are happy to use a different shuffle algo if it's faster.

Reply:
I wanted to preserve the defaults that worked for you guys in the past, so was keeping `py1s`. `py1e` won't be faster but it'll give you a better shuffle. Since the shuffle is stronger though, it could lead to some slowdowns in some cases, depending on the size of the dataset samples. Because your datasets are so large I think having `py1s` as the default and setting `py1e` if you'd like a stronger shuffle is better.

Reply:
lmk if any other changes or if you think switching to `py1e` is best.

Reply:
Oh what would be better but give us a decent shuffle?

Reply:
If you'd like a better shuffle then setting `py1e` in the yaml should be fine. But jut going to leave this as the default for now.